src/qt/addresstablemodel.cpp

```diff
@@ -97,39 +102,22 @@ class AddressTablePriv
         {
             LOCK(wallet->cs_wallet);
             for (const PAIRTYPE(CTxDestination, AddressBook::CAddressBookData) & item : wallet->mapAddressBook) {
-                const CBitcoinAddress& address = item.first;
+
+                const CChainParams::Base58Type addrType =
+                        AddressBook::IsColdStakingPurpose(item.second.purpose) ?
+                        CChainParams::STAKING_ADDRESS : CChainParams::PUBKEY_ADDRESS;
+                const CBitcoinAddress address = CBitcoinAddress(item.first, addrType);
```

Comment:
purposes `DELEGABLE` and `DELEGATOR` give `IsColdStakingPurpose` true, but their addresses should be encoded as standard addresses (with `CChainParams::PUBKEY_ADDRESS` base58 prefix).

Reply:
good catch